src/main/java/uk/gov/hmcts/reform/et/syaapi/service/CaseDocumentService.java

```diff
@@ -0,0 +1,211 @@
+package uk.gov.hmcts.reform.et.syaapi.service;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.tika.Tika;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.et.syaapi.models.CaseDocument;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static uk.gov.hmcts.reform.ccd.client.model.Classification.PUBLIC;
+
+/**
+ * CaseDocumentService provides access to the document upload service API, used to upload documents that are
+ * associated to a specific case record held in CCD.
+ * <p/>
+ * This relies upon the following configurations to be set at an environment level:
+ * <ul>
+ *     <li>CASE_DOCUMENT_AM_URL</li>
+ * </ul>
+ */
+@Slf4j
+@Service
+public class CaseDocumentService {
+
+    private static final String JURISDICTION = "EMPLOYMENT";
+    private static final String SERVICE_AUTHORIZATION = "ServiceAuthorization";
+    private static final String FILE_NAME_REGEX_PATTERN = "^[\\w\\- ]{1,256}+\\.[A-Za-z]{3,4}$";
+    private static final Pattern FILE_NAME_PATTERN = Pattern.compile(FILE_NAME_REGEX_PATTERN);
+    private static final String HTTPS_URL_REGEX_PATTERN =
+        "^(http://www\\.|https://www\\.|http://|https://)?[\\w\\-]+(\\.[a-z]{2,5})?(:\\d{1,5})?(/.*)?$";
+    private static final Pattern HTTPS_URL_PATTERN = Pattern.compile(HTTPS_URL_REGEX_PATTERN);
+    private static final String UPLOAD_FILE_EXCEPTION_MESSAGE = "Document management failed uploading file: ";
+    private static final String VALIDATE_FILE_EXCEPTION_MESSAGE = "File does not pass validation";
+    public static final int MAX_API_RETRIES = 3;
```

Comment:
If you're going to use this, then better to @Value it in the constructor with a default value of 3.  Then document the configuration at top.  But see comment further down first as it may not be relevant

Reply:
applied